core/domain/classifier_services_test.py

```diff
@@ -0,0 +1,166 @@
+
+    def test_valid_state(self):
+        self.assertEquals(self.string_classifier._label_count, 3)
+        self.assertEquals(self.string_classifier._doc_count, 3)
+        self.assertEquals(self.string_classifier._word_count, 10)
+        self._validate_instance(self.string_classifier)
+
+    def test_add_examples(self):
+        self.string_classifier.add_examples(self._NEW_EXAMPLES)
+        self.assertEquals(self.string_classifier._label_count, 3)
+        self.assertEquals(self.string_classifier._doc_count, 6)
+        self.assertEquals(self.string_classifier._word_count, 37)
+        self._validate_instance(self.string_classifier)
+
+    def test_empty_load(self):
+        self.string_classifier.load_examples([])
+
+    def test_empty_add(self):
+        self.string_classifier.add_examples([])
+
+    def test_model_save_load(self):
+        self.assertEquals(
+            self.string_classifier._doc_count,
+            len(self._EXAMPLES))
+        model = self.string_classifier.to_dict()
+        model['_doc_count'] = 9
+        self.assertEquals(model['_doc_count'], 9)
+        self.assertEquals(
+            self.string_classifier._doc_count,
+            len(self._EXAMPLES))
+        self.string_classifier.add_examples(self._NEW_EXAMPLES)
+        self.assertEquals(
+            self.string_classifier._doc_count,
+            len(self._EXAMPLES) + len(self._NEW_EXAMPLES))
+        self.assertEquals(model['_doc_count'], 9)
+        self.string_classifier.from_dict(model)
+        self.assertEquals(self.string_classifier._doc_count, 9)
+        self.assertEquals(model['_doc_count'], 9)
+
+    def test_get_word_id(self):
+        word_count = self.string_classifier._word_count
+        self.string_classifier._get_word_id('_non_existent_word_1')
+        self.assertEquals(self.string_classifier._word_count, word_count + 1)
+        self.string_classifier._get_word_id('i')
+        self.assertEquals(self.string_classifier._word_count, word_count + 1)
+        self.string_classifier._get_word_id('_non_existent_word_2')
+        self.assertEquals(self.string_classifier._word_count, word_count + 2)
+
+    def test_get_label_id(self):
+        label_count = self.string_classifier._label_count
+        self.string_classifier._get_label_id('_non_existent_label_1')
+        self.assertEquals(self.string_classifier._label_count, label_count + 1)
+        self.string_classifier._get_label_id('food')
+        self.assertEquals(self.string_classifier._label_count, label_count + 1)
+        self.string_classifier._get_label_id('_non_existent_label_2')
+        self.assertEquals(self.string_classifier._label_count, label_count + 2)
+
+    def test_label_begins_with_underscore(self):
+        self.string_classifier.add_examples([['example doc', ['good_label']]])
+        with self.assertRaises(Exception):
+            self.string_classifier.add_examples(
+                [['example doc', ['_bad_label']]])
+
+    def test_reload_valid_state(self):
```

Comment:
I believe that is test_empty_load and test_empty_add, but I can pin more sanity checks to them.